### ISSUE TITLE
Incremental Loads fail due to Key Formatting

### ISSUE THREAD

**Issue by isaiahgm** [wintersrd/pipelinewise-tap-mssql#67]
I've run into the following error when running Incremental jobs utilizing this tap:
```
pymssql._mssql.MSSQLDatabaseException: (241, b'Conversion failed when converting date and/or time from character string.DB-Lib error message 20018, severity 16:\nGeneral SQL Server error: Check messages from the SQL Server\n')
```
The issue can be solved by adding a string format to the replication_key_value [here](https://github.com/wintersrd/pipelinewise-tap-mssql/blob/master/tap_mssql/sync_strategies/incremental.py#L62) so it is always sent to SQL Server in a format that it can automatically parse like so:
```
params["replication_key_value"] = replication_key_value.format('YYYY-MM-DDTHH:MM:ss.SSS')
``` 
However, this seems like is a somewhat unsophisticated approach, and it cuts out timezone information from the state. I'm open to other solutions, but will attach a MR shortly with this proposed change.

**Comment by isaiahgm** [comment]
#68

**Comment by mjsqu** [comment]
I have this problem now too, it appears to have worked as recently as 5th April 2024 - Just looking at the changelog for pymssql: https://github.com/pymssql/pymssql/blob/v2.3.0/ChangeLog.rst#version-230---2024-04-06---mikhail-terekhov

The update on 6th April to pymssql moving it to 2.3.0 may have broken things for us. I will look into this today as it'll be a blocker on something I'm working on.

**Comment by mjsqu** [comment]
Thank you for raising this @isaiahgm. After a fairly deep dive into this, I think the problem lies in the new version of pymssql not accepting Pendulum datetime values: I raised an issue over there https://github.com/pymssql/pymssql/issues/889

I picked up a fix from this StackOverflow page: https://stackoverflow.com/questions/54028325/how-to-convert-pendulum-to-datetime-datetime-type

A suggested fix that keeps the python variable in datetime.datetime format would be:

```
datetime.fromtimestamp(replication_key_value.timestamp())
```

**Comment by isaiahgm** [comment]
Hey @mjsqu, thanks for digging into this! I'll keep an eye on that new issue, but I like this proposed workaround more than what I threw together. It seems like the ideal path is to update pymssql to support Pendulum again then update this project to use the newer package version, but for the meantime I'll implement this workaround for my projects.

**Comment by s7clarke10** [comment]
Hi @isaiahgm,I have discussed this change with @mjsqu. He has prepared a change as he has described for review. Included in the change is a description for the issue and bumping the versions.I hope to review and push this change through tomorrow.On 18 Apr 2024, at 3:52 PM, Isaiah Morgan ***@***.***> wrote:﻿
Hey @mjsqu, thanks for digging into this! I'll keep an eye on that new issue, but I like this proposed workaround more than what I threw together. It seems like the ideal path is to update pymssql to support Pendulum again then update this project to use the newer package version, but for the meantime I'll implement this workaround for my projects.

**Comment by isaiahgm** [comment]
Sounds great! Thank you @s7clarke10 and @mjsqu for all your help on this.

**Comment by s7clarke10** [comment]
Fixed by #69 . Thank you for raising this issue with a suggested change.
